### PR TITLE
enhance Mac keymap

### PR DESCRIPTION
**Pull request by wicksome**

### CODE REVIEW

File: keymaps/intellij-idea-keymap.cson [hunk 1]
```diff
@@ -28,6 +28,24 @@
...
 ####################################################################################################
 
+'body':
+
+  ##########
+  # window #
+  ##########
+
+  # Select Next Tab
+  'cmd-shift-]': 'pane:show-next-item'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: This shortcut already works out of the box in Atom. I would remove it here such that we just have the shortcuts which are different between IDEA and Atom.

File: keymaps/intellij-idea-keymap.cson [hunk 2]
```diff
@@ -28,6 +28,24 @@
...
+
+  # Select Previous Tab
+  'cmd-shift-[': 'pane:show-previous-item'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: This shortcut already works out of the box in Atom. I would remove it here such that we just have the shortcuts which are different between IDEA and Atom.

File: keymaps/intellij-idea-keymap.cson [hunk 3]
```diff
@@ -113,13 +140,54 @@
...
   # Insert live templates
   'cmd-j': 'snippets:available'
 
+'.platform-darwin .tree-view':
+
+  ###########
+  # General #
+  ###########
+
+  'left': 'tree-view:collapse-directory'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: This shortcut already works out of the box in Atom. I would remove it here such that we just have the shortcuts which are different between IDEA and Atom.

This is also the case for `right` and `enter`.

File: keymaps/intellij-idea-keymap.cson [hunk 4]
```diff
@@ -113,13 +140,54 @@
...
+  ##############
+
+  # Open corresponding tool window
+  'cmd-1': 'tree-view:toggle'
+
+  ###########
+  # Editing #
+  ###########
+
+  'cmd-c': 'tree-view:copy'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: This shortcut already works out of the box in Atom. I would remove it here such that we just have the shortcuts which are different between IDEA and Atom.

This is also the case for:
- `cmd-x`
- `cmd-v`
- `backspace`

The `delete` I cannot test as I only have the laptop keyboard. ;)

**Review by guylabs — CHANGES_REQUESTED** [review]
Hi @5pid 

thanks a lot for the PR! Really appreciate such contributions. I just added some review comments. Could you maybe go through them and adapt these such that I can merge it with the master and release a new version?

If you have any questions please ask.

Thanks and regards,

Guy

**Comment by wicksome** [comment]
@guylabs 
Thanks for reviews.
I fixed some shortcuts that you refered to me.
Please check commit.

File: keymaps/intellij-idea-keymap.cson (hunk 1 above)
Comment (wicksome) [reply keymaps/intellij-idea-keymap.cson]: deleted line.

Comment (guylabs) [reply keymaps/intellij-idea-keymap.cson]: This line is still in and is not deleted. Could you also delete this one?

File: keymaps/intellij-idea-keymap.cson (hunk 2 above)
Comment (guylabs) [reply keymaps/intellij-idea-keymap.cson]: This line is still in and is not deleted. Could you also delete this one?

File: keymaps/intellij-idea-keymap.cson [hunk 5]
```diff
@@ -109,17 +136,55 @@
 
   # Go to line
   'cmd-l': 'go-to-line:toggle'
+  'cmd-\\': 'unset!' # cancel the tree-view shortcut
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: I would not unset the standard shortcuts as the shortcuts for idea should just be an addition to the default ones. Could you maybe remove all the `unset!` shortcuts?

File: keymaps/intellij-idea-keymap.cson [hunk 6]
```diff
@@ -109,17 +136,55 @@
...
+  ##############
+
+  # Open corresponding tool window
+  'cmd-1': 'tree-view:toggle'
+
+  ###########
+  # Editing #
+  ###########
+
+  'ctrl-shift-C': 'unset!'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: I would not unset the standard shortcuts as the shortcuts for idea should just be an addition to the default ones. Could you maybe remove all the `unset!` shortcuts?

File: keymaps/intellij-idea-keymap.cson [hunk 7]
```diff
@@ -109,17 +136,55 @@
...
+  'cmd-shift-c': 'tree-view:copy-full-path'
+  'f2': 'unset!'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: I would not unset the standard shortcuts as the shortcuts for idea should just be an addition to the default ones. Could you maybe remove all the `unset!` shortcuts?

File: keymaps/intellij-idea-keymap.cson [hunk 8]
```diff
@@ -109,17 +136,55 @@
 
   # Go to line
   'cmd-l': 'go-to-line:toggle'
+  'cmd-\\': 'unset!' # cancel the tree-view shortcut
+  'cmd-k cmd-b': 'unset!' # cancel the tree-view shortcut
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: I would not unset the standard shortcuts as the shortcuts for idea should just be an addition to the default ones. Could you maybe remove all the `unset!` shortcuts?

File: keymaps/intellij-idea-keymap.cson [hunk 9]
```diff
@@ -109,17 +136,55 @@
...
   # Insert live templates
   'cmd-j': 'snippets:available'
 
+'.platform-darwin .tree-view':
+
+  ###########
+  # General #
+  ###########
+
+  'a': 'unset!'
```
Comment (guylabs) [inline keymaps/intellij-idea-keymap.cson]: I would not unset the standard shortcuts as the shortcuts for idea should just be an addition to the default ones. Could you maybe remove all the `unset!` shortcuts?

**Review by guylabs — CHANGES_REQUESTED** [review]
Hi @5pid 

thanks for the quick changes but there are still some lines which need to be removed and please also remove all the `unset!` shortcuts as we do not want to remove the standard Atom shortcuts.

After these changes I can merge it and release a newer version.

Thanks and regards,

Guy

File: keymaps/intellij-idea-keymap.cson (hunk 1 above)
Comment (wicksome) [reply keymaps/intellij-idea-keymap.cson]: Sorry. It is my mistake.

**Comment by wicksome** [comment]
@guylabs 
Sorry. my mistake.
I deleted that missing codes.
Thanks for reviews.

**Review by guylabs — APPROVED** [review]

**Pull request merged**